caikit/interfaces/nlp/data_model/classification.py

```diff
@@ -139,6 +152,6 @@ class ClassifiedGeneratedTextStreamResult(ClassifiedGeneratedTextResult):
     """
 
     processed_index: Annotated[
-        Optional[int], FieldNumber(7)
+        Optional[int], FieldNumber(8)
```

Comment:
I know this probably isn't gonna break anybody if these interfaces are only used in an http server, but `FieldNumber` here maps to a protobuf field number, which cannot be changed without breaking the wire format serialization. See: https://github.com/caikit/caikit/pull/568/checks?check_run_id=18419777460

Reply:
Hah, and this case is called out specifically 😉 

> Common causes of field number reuse:
> - renumbering fields (sometimes done to achieve a more aesthetically pleasing number order for fields). Renumbering effectively deletes and re-adds all the fields involved in the renumbering, resulting in incompatible wire-format changes.
> - deleting a field and not [reserving](https://protobuf.dev/programming-guides/proto3/#fieldreserved) the number to prevent future reuse.

Reply:
Oh, I was concerned about renumbering, but wasn't sure it would break anything since it still worked in the guardrails test.. In that case, if I add the warnings field number to be a new field number that doesn't disrupt the order, will that suffice or still cause issues @joerunde ?

Reply:
Yeah you'll need to make sure to not change any existing numbers, and new fields need to use a number that isn't already in use. You would be able to see this fail if you ran a grpc server without the changes and a grpc client with your new changes. Since guardrails is http only, you probably wouldn't even be able to trigger the failure, but we should still make sure we understand how the field numbering works and practice not breaking it.